compiler/modules/bir/bir.bal

```diff
@@ -92,6 +97,22 @@ public type FunctionSignature readonly & record {|
     SemType? restParamType = ();
 |};
 
+public type Region record {|
+    Label entry;
+    Label? exit = ();
+    RegionIndex? parent = ();
+    RegionKind kind;
+|};
+
+public enum RegionKind {
+    # Region that represents a loop with LoopHeader as the entry
+    REGION_LOOP,
+    # Region that has an entry which diverge into several possible branches, that eventually rejoin.
```

Comment:
Describe in BIR terms: i.e. has a CondBranch insn

Reply:
Since regions are single exit, they have to rejoin: no need to say that.

I think REGION_COND is a better name.